[Signal Store] Access the whole state object

### Which @ngrx/* package(s) are relevant/related to the feature request?

signals

### Information

Using `@ngrx/signals` 17.0.0-beta.0

It doesn't look like it's currently possible to access the whole state object from a `signalStore` instance as a single signal.

It would be useful to be able to do so, e.g. in cases where the state object is modelled as a state machine using type unions and a discriminated field.

I've noticed the autocompletion does allow you to do something like:

```ts
import { STATE_SIGNAL } from '@ngrx/signals/src/signal-state';
...
vm = this.#store[STATE_SIGNAL];
```

… which compiles fine, but at runtime there is an error:

```shell
✘ [ERROR] Could not resolve "@ngrx/signals/src/signal-state"

    src/app/login/feature/login-page.component.ts:3:29:
      3 │ import { STATE_SIGNAL } from '@ngrx/signals/src/signal-state';
        ╵                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  The path "./src/signal-state" is not exported by package "@ngrx/signals":

    node_modules/@ngrx/signals/package.json:55:13:
      55 │   "exports": {
         ╵              ^

  You can mark the path "@ngrx/signals/src/signal-state" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.
```

Note: it would also be very useful to access this single signal in the `withMethods`, `withHooks` etc. A concrete use case is setting up an `effect` in `onInit` to log state changes in the whole state object.

### Describe any alternatives/workarounds you're currently using

No simple working alternatives found.

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

--- a/modules/signals/spec/get-state.spec.ts
+++ b/modules/signals/spec/get-state.spec.ts
@@ -1,0 +1,53 @@
+import { TestBed } from '@angular/core/testing';
+import {
+  getState,
+  patchState,
+  signalState,
+  signalStore,
+  withState,
+} from '../src';
+import { effect } from '@angular/core';
+
+describe('getState', () => {
+  const initialState = {
+    user: {
+      firstName: 'John',
+      lastName: 'Smith',
+    },
+    foo: 'bar',
+    numbers: [1, 2, 3],
+    ngrx: 'signals',
+  };
+
+  describe('with signalStore', () => {
+    it('returns the state object', () => {
+      const Store = signalStore(withState(initialState));
+      const store = new Store();
+
+      expect(getState(store)).toEqual(initialState);
+    });
+
+    it('executes in the reactive context', () => {
+      const Store = signalStore(withState(initialState));
+      const store = new Store();
+
+      let executionCount = 0;
+
+      TestBed.runInInjectionContext(() => {
+        effect(() => {
+          getState(store);
+          executionCount++;
+        });
+      });
+
+      TestBed.flushEffects();
+      expect(executionCount).toBe(1);
+
+      patchState(store, { foo: 'baz' });
+
+      TestBed.flushEffects();
+      expect(executionCount).toBe(2);
+      expect(getState(store)).toEqual({ ...initialState, foo: 'baz' });
+    });
+  });
+});

--- a/modules/signals/src/get-state.ts
+++ b/modules/signals/src/get-state.ts
@@ -1,0 +1,7 @@
+import { STATE_SIGNAL, SignalStateMeta } from './signal-state';
+
+export function getState<State extends Record<string, unknown>>(
+  signalState: SignalStateMeta<State>
+): State {
+  return signalState[STATE_SIGNAL]();
+}

--- a/modules/signals/src/index.ts
+++ b/modules/signals/src/index.ts
@@ -1,3 +1,4 @@
+export { getState } from './get-state';
 export { PartialStateUpdater, patchState } from './patch-state';
 export { signalState } from './signal-state';
 export { signalStore } from './signal-store';

